### PR TITLE
Add case_sensitive attr to FilteredList

### PR DESCRIPTION
**Pull request by robob27**
Add a `case_sensitive` attr to `FilteredList` to allow for case insensitive matching. Defaults to `true` so that existing `FilteredList`s keep the same behavior.

### CODE REVIEW

File: library/lua/gui/widgets.lua [hunk 1]
```diff
@@ -2026,11 +2027,17 @@ function FilteredList:setFilter(filter, pos)
...
                         not search_key:match('%f[^%s\x00]'..key) then
-                    ok = false
-                    break
+                            ok = false
+                            break
+                    elseif not string.lower(search_key):match('%f[^%p\x00]'..string.lower(key)) and
+                        not string.lower(search_key):match('%f[^%s\x00]'..string.lower(key)) then
+                            ok = false
+                            break
+                    end
```
Comment (myk002) [inline library/lua/gui/widgets.lua:2040]: could we factor out the lowercasing and leave the rest of the code un-duplicated? every time a regular expression is duplicated in code, a magical forest creature dies.

**Review by myk002 — COMMENTED** [review]
it would be very nice to have the match be case insensitive by default, but we need to check all users of the widget for negative interactions first. I'm happy merging this with default case sensitivity now, but once we have time to check the (in-tree) users of List for negative interactions, we can switch the default value to false.

also, changelog plz

**Review by myk002 — COMMENTED** [review]
looks good; just needs a changelog entry

**Comment by robob27** [comment]
> looks good; just needs a changelog entry

Would it go in the future section currently?

**Comment by myk002** [comment]
Yes, all new changelog entries go in "Future"

**Review by myk002 — APPROVED** [review]

**Pull request merged**